pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -600,6 +601,29 @@ public void destroy() {
     return intIterators;
   }
 
+  private IntIterator[] getSortedBitmapIntIteratorsForBytesColumn(String column) {
+    MutableDictionary dictionary = _dictionaryMap.get(column);
+    int numValues = dictionary.length();
+    IntIterator[] intIterators = new IntIterator[numValues];
+    RealtimeInvertedIndexReader invertedIndex = _invertedIndexMap.get(column);
+
+    ByteArray[] values = new ByteArray[numValues];
```

Comment:
can we not sort it without creating a copy? use a comparator which will get byte[] and you can use convert it to ByteArray[].

Reply:
This is to follow the pattern of other types. ByteArray will only take byte[] ref but not copy the content.